### DISCUSSION TITLE
What is the best way to store container logs in file with rotating?

### DISCUSSION THREAD

**Discussion by xluwo** [containers/podman#21141] [Q&A]
Hi I want to store my rootless container logs in file with rotating. What is the best way to achieve that?

This is my current setup:
```
    --log-driver k8s-file
    --log-opt path=/var/log/{{servicename}}/server.log
    --log-opt max-size=100mb
```

I also have logrotate configuration to rotate `/var/log/{{servicename}}/server.log` file:
```
/var/log/{{servicename}}/*log {
    su {{servicename}} {{servicename}}
    daily
    copytruncate
    rotate 90
    missingok
    notifempty
}
```

The problems I see in my current setup:
1. Sometimes server.log is converted to binary file (not sure why maybe you know the reason?). To browse it I have to use `strings server.log`.
2. I rotate logs daily which means I can loose some logs if file gets larger than 100mb. Is there any way to setup logs rotating using podman log-opt option without logrotate?

**Comment by ankit-4129** [comment]
@xluwo 

> I rotate logs daily which means I can loose some logs if file gets larger than 100mb. Is there any way to setup logs rotating using podman log-opt option without logrotate?

if you have any solution or workaround for it can you share it
